src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategyGuarantee.kt

```diff
@@ -77,7 +77,35 @@ class ManagedStrategyGuarantee private constructor(
     }
 }
 
+@Suppress("UNUSED_PARAMETER")
+internal fun getDefaultSilentSectionGuarantee(className: String, methodName: String): ManagedGuaranteeType? {
+    if (className.startsWith("java.util.concurrent.")) {
+        if (isJavaExecutorService(className)) {
+            if (methodName == "submit") {
+                return ManagedGuaranteeType.SILENT_NESTED
+            }
+            return ManagedGuaranteeType.SILENT
+        }
+        if (className.startsWith("java.util.concurrent.locks.AbstractQueuedSynchronizer"))
+            return ManagedGuaranteeType.SILENT
+        if (className == "java.util.concurrent.FutureTask")
+            return ManagedGuaranteeType.SILENT
+    }
+    return null
+}
+
+private fun isJavaExecutorService(className: String) =
+    className.startsWith("java.util.concurrent.AbstractExecutorService") ||
+    className.startsWith("java.util.concurrent.ThreadPoolExecutor") ||
+    className.startsWith("java.util.concurrent.ForkJoinPool")
+
 internal enum class ManagedGuaranteeType {
     IGNORE,
-    ATOMIC
-}
\ No newline at end of file
+    ATOMIC,
+    SILENT,
+    SILENT_NESTED,
```

Comment:
What is the difference between "silent" and "silent nested"? Could you please document the modes?

Reply:
Added documentation that explains the difference between various sections.